BHoM_Engine/Create/CustomObject.cs

```diff
@@ -44,21 +45,89 @@ public static CustomObject CustomObject(Dictionary<string, object> data, string
 
         public static CustomObject CustomObject(List<string> propertyNames, List<object> propertyValues, string name = "")
         {
-            Dictionary<string, object> customData = new Dictionary<string, object>();
+            CustomObject result = new CustomObject();
 
             if (propertyNames.Count == propertyValues.Count)
             {
                 for (int i = 0; i < propertyValues.Count; i++)
-                    customData.Add(propertyNames[i], propertyValues[i]);
+                {
+                    string propName = propertyNames[i];
+                    object propValue = propertyValues[i];
+
+                    if (propName == "Name" && propValue is string)
+                        result.Name = propValue as string;
+                    else if (propName == "BHoM_Guid" && propValue is System.Guid)
+                        result.BHoM_Guid = (System.Guid)propValue;
+                    else if (propName == "Tags")
+                        SetTags(result, propValue as dynamic);
+                    else if (propName == "Fragments")
+                        SetFragments(result, propValue as dynamic);
+                    else
+                        result.CustomData.Add(propName, propValue);
+                }     
             }
             else
                 throw new System.Exception("The list of property names must be the same length as the list of property values when creating a Custon object.");
 
-            return new CustomObject
-            {
-                CustomData = customData,
-                Name = name
-            };
+            if (name != "")
```

Comment:
Shouldn't it be `if (!string.IsNullOrWhiteSpace(name))`?

Reply:
It is safer, yes. Now corrected.